packages/react-core/src/components/NumberInput/examples/NumberInput.md

```diff
@@ -37,6 +37,10 @@ class BasicNumberInput extends React.Component {
         value: this.state.value + 1
       });
     };
+
+    this.onBlur = (e) => {
+      e.target.value = Number(e.target.value).toString();
```

Comment:
It looks like we want to directly update the `value` state variable - this aligns with the implementation of all the other event handlers. 

Can we use state wherever `this.onBlur` is declared in `NumberInput.md`? 

```suggestion
      const newValue = Number(e.target.value).toString();
       this.setState({
         value: newValue
       });
```

Reply:
Good catch! I'll add that to all of the `onBlur` instances :)

Reply:
We might not need to update the state in `onBlur` since we do that in `onChange` and console logging `value` in both `onChange` and `onBlur` appear to have the leading zero cut. I'm going to keep looking into this!

Reply:
Ah you're right! Did some investigating here too... since we need to keep the `toString()` for the leading zeros fix, I'm OK approving as-is -- matches how @jschuler wrote it in #7117 👍

Reply:
Sounds great! I just made sure that `onBlur` was also added to the second example with the monetary amount